spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java

```diff
@@ -177,27 +170,30 @@ CurrentTraceContext currentTraceContext() {
 		@Nullable
 		final TraceContext parent;
 
-		private final Span span;
-
 		MonoWebClientTrace(ExchangeFunction next, ClientRequest request,
-				TraceExchangeFilterFunction filterFunction, @Nullable TraceContext parent,
-				Span span) {
+				TraceExchangeFilterFunction filterFunction,
+				@Nullable TraceContext parent) {
 			this.next = next;
 			this.request = request;
 			this.handler = filterFunction.handler();
 			this.currentTraceContext = filterFunction.currentTraceContext();
 			this.scopePassingTransformer = filterFunction.scopePassingTransformer;
 			this.parent = parent;
-			this.span = span;
 		}
 
 		@Override
 		public void subscribe(CoreSubscriber<? super ClientResponse> subscriber) {
 
 			Context context = subscriber.currentContext();
 
-			this.next.exchange(request).subscribe(new WebClientTracerSubscriber(
-					subscriber, context, parent, span, this));
+			HttpClientRequest wrapper = new HttpClientRequest(request);
+			Span span = handler.handleSendWithParent(wrapper, parent);
+			if (log.isDebugEnabled()) {
+				log.debug("HttpClientHandler::handleSend: " + span);
+			}
+
+			this.next.exchange(wrapper.buildRequest()).subscribe(
```

Comment:
it is my understanding that this `.exchange()` starts the request, so we should do the `handleSend()` stuff here as that's right before the request is made. correct?